code/datums/craft/gun_parts.dm

```diff
@@ -21,6 +21,40 @@
 	var/barrel = /obj/item/part/gun/barrel
 	var/barrel_attached = FALSE
 
+/obj/item/part/gun/frame/New(loc)
+	..()
+	var/spawn_with_preinstalled_parts = FALSE
+	if(istype(loc, /obj/structure/scrap_spawner))
+		spawn_with_preinstalled_parts = TRUE
+	else if(in_maintenance())
+		var/turf/T = get_turf(src)
+		for(var/atom/A in T.contents)
+			if(istype(A, /obj/spawner))
+				spawn_with_preinstalled_parts = TRUE
```

Comment:
This is very cursed but okay.

Reply:
I couldn't figure out an elegant way to make only the maint and trash pile spawns have the extra parts without creating infinite part exploits.